contracts/core/ManualVolOracle.sol

```diff
@@ -0,0 +1,280 @@
+     */
+    constructor(uint32 _period) {
+        require(_period > 0, "!_period");
+
+        period = _period;
+
+        // 31536000 seconds in a year
+        // divided by the period duration
+        // For e.g. if period = 1 day = 86400 seconds
+        // It would be 31536000/86400 = 365 days.
+        annualizationConstant = Math.sqrt(uint256(31536000).div(_period));
+
+        // Add beneficiary as admin
+        _setupRole(ADMIN_ROLE, msg.sender);
+    }
+
+    /// @dev A modifier which checks that the caller has the admin role.
+    modifier onlyAdmin() {
+        require(hasRole(ADMIN_ROLE, msg.sender), "VolOracle: only admin");
+        _;
+    }
+
+    /**
+     * @notice Commits an oracle update
+     */
+    function commit(address pool) external {
+        (uint32 commitTimestamp, uint32 gapFromPeriod) = secondsFromPeriod();
+        require(gapFromPeriod < commitPhaseDuration, "Not commit phase");
+
+        uint256 price = twap(pool);
+        uint256 _lastPrice = lastPrices[pool];
+        uint256 periodReturn = _lastPrice > 0 ? wdiv(price, _lastPrice) : 0;
+
+        // logReturn is in 10**18
+        // we need to scale it down to 10**8
+        int256 logReturn =
+            periodReturn > 0
+                ? PRBMathSD59x18.ln(int256(periodReturn)) / 10**10
+                : 0;
+
+        Accumulator storage accum = accumulators[pool];
+
+        require(
+            block.timestamp >=
+                accum.lastTimestamp + period - commitPhaseDuration,
+            "Committed"
+        );
+
+        (uint256 newCount, int256 newMean, uint256 newM2) =
+            Welford.update(accum.count, accum.mean, accum.m2, logReturn);
+
+        require(newCount < type(uint16).max, ">U16");
+        require(newMean < type(int96).max, ">I96");
+        require(newM2 < type(uint112).max, ">U112");
+
+        accum.count = uint16(newCount);
+        accum.mean = int96(newMean);
+        accum.m2 = uint112(newM2);
+        accum.lastTimestamp = commitTimestamp;
+        lastPrices[pool] = price;
+
+        emit Commit(
+            uint16(newCount),
+            uint32(commitTimestamp),
+            int96(newMean),
+            uint112(newM2),
+            price,
+            msg.sender
+        );
+    }
+
+    /**
+     * @notice Returns the standard deviation of the base currency in 10**8 i.e. 1*10**8 = 100%
+     * @return standardDeviation is the standard deviation of the asset
+     */
+    function vol(address pool) public view returns (uint256 standardDeviation) {
+        return Welford.stdev(accumulators[pool].count, accumulators[pool].m2);
+    }
+
+    /**
+     * @notice Returns the annualized standard deviation of the base currency in 10**8 i.e. 1*10**8 = 100%
+     * @return annualStdev is the annualized standard deviation of the asset
+     */
+    function annualizedVol(address pool)
+        public
+        view
+        returns (uint256 annualStdev)
+    {
+        return annualizedVols[pool];
```

Comment:
do we want to pass in already annualized or should we annualize it in here

Reply:
Let's set the annualized vol number directly